ALVRClient/Renderer.swift

```diff
@@ -371,13 +423,20 @@ class Renderer {
                 hudMessageBuffer.deallocate()
             case ALVR_EVENT_STREAMING_STARTED.rawValue:
                 print("streaming started: \(alvrEvent.STREAMING_STARTED)")
-                let foveationVars = FFR.calculateFoveationVars(alvrEvent.STREAMING_STARTED)
+                let settingsJsonBuffer = UnsafeMutableBufferPointer<CChar>.allocate(capacity: 1024)
+                alvr_get_settings_json(settingsJsonBuffer.baseAddress)
+                let settingsJsonStr = String(cString: settingsJsonBuffer.baseAddress!, encoding: .utf8)!
+                let settingsData = settingsJsonStr.data(using: .utf8)!
```

Comment:
Rather than creating a String and turning it into Data, create the Data directly. The buffer also needs to be dealloc'd.

```suggestion
                let settingsData = Data(buffer: settingsJsonBuffer)
                settingsJsonBuffer.deallocate()
```

Or, if Rust allocates the buffer and returns a void* and count, you could use `Data(bytesNoCopy:count:deallocator:)`.

Reply:
Thanks for the explanation! I allocated a buffer in Swift, populated it from Rust, and then used  `Data(bytesNoCopy:count:deallocator:)` to wrap it.

Seems like there is a preference for defining deallocation near the definition of the `UnsafePointer`, so I've opted for that instead of using a deallocator.